main.cpp

```diff
@@ -37,8 +37,89 @@ send me a DM to check your pull request
  Wait for my code review.
  */
 
+struct FloatType
+{
+    float add(float lhs, float rhs)
+    {
+        return lhs + rhs;
+    }
+    float subtract(float lhs, float rhs)
+    {
+        return lhs - rhs;
+    }    
+    float multiply(float lhs, float rhs)
+    {
+        return lhs * rhs;
+    }
+    float divide(float lhs, float rhs)
+    {
+        return lhs / rhs;
+    }
+};
+
+struct DoubleType
+{
+    double add(double lhs, double rhs)
+    {
+        return lhs + rhs;
+    }
+    double subtract(double lhs, double rhs)
+    {
+        return lhs - rhs;
+    }    
+    double multiply(double lhs, double rhs)
+    {
+        return lhs * rhs;
+    }
+    double divide(double lhs, double rhs)
+    {
+        return lhs / rhs;
+    }
+};
+
+struct IntType
+{
+    int add(int lhs, int rhs)
+    {
+        return lhs + rhs;
+    }
+    int subtract(int lhs, int rhs)
+    {
+        return lhs - rhs;
+    }    
+    int multiply(int lhs, int rhs)
+    {
+        return lhs * rhs;
+    }
+    int divide(int lhs, int rhs)
+    {
+        if(rhs == 0) return 0;
+        else return lhs / rhs;
```

Comment:
no need for the 'else'!
if `rhs == 0`, execution ends at `return 0`.  `if rhs != 0`, that line is skipped. 
- [ ] remove `else` keyword